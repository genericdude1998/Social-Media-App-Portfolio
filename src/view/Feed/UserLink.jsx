import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './NavBar.css';


const UserLink = ({id, name, className}) => {
    return (
        <NavLink to={`/userInfo/${id}`} className={(navData) => `${className} ${navData.isActive ? styles.active : null}`}>
            {name}
        </NavLink>
    );
}

UserLink.propTypes = {
    id:PropTypes.number,
    name:PropTypes.string,
    className:PropTypes.string,
}

export default UserLink;
