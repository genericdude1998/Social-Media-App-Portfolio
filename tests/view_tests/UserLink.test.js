import React from 'react'
import { shallow } from 'enzyme';
import UserLink from '../../src/view/Feed/UserLink';
import {mockId, mockName} from '../mockValues';
import { NavLink } from 'react-router-dom';
import {mockActiveNavData, mockInActiveNavData} from '../mockValues';

let wrapper = shallow(
    <UserLink 
        id={mockId}
        name={mockName}
        className={'mockClass'}
    />
)

describe('UserLink', () => {
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should have active styles', () => {
        const navLinkClassnameFunc  =  wrapper.find(NavLink).prop('className');
        expect(navLinkClassnameFunc(mockActiveNavData)).toContain('active');
    });
    it('should have inactive styles', () => {
        const navLinkClassnameFunc  =  wrapper.find(NavLink).prop('className');
        expect(navLinkClassnameFunc(mockInActiveNavData)).toContain('null');
    });
});