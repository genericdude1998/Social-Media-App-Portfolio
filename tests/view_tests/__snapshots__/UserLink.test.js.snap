// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLink should match snapshot 1`] = `
<NavLink
  className={[Function]}
  to="/userInfo/0"
>
  mock_name
</NavLink>
`;
